sql/sql_select.cc

```diff
@@ -16705,6 +16707,20 @@ create_tmp_table(THD *thd, TMP_TABLE_PARAM *param, List<Item> &fields,
 	  goto err;				// Got OOM
 	continue;				// Some kind of const item
       }
+
+      if (type == Item::FIELD_ITEM)
+      {
+        Item_field *item_field= (Item_field *)item;
+        Field *field= item_field->field;
+        TABLE_SHARE *s= field->table->s;
+        if (s->versioned)
+        {
+          if (field->flags & VERS_SYS_START_FLAG && fieldnr == field_count - 2)
+            sys_trx_start= new_field;
+          if (field->flags & VERS_SYS_END_FLAG && fieldnr == field_count - 1)
+            sys_trx_end= new_field;
```

Comment:
Don't repeat same condition (see comment for 377960e36353a2af02e2e8470de32d252f011f0c).